odl/operator/pspace_ops.py

```diff
@@ -26,8 +26,8 @@
 import numpy as np
 import scipy as sp
 
-from odl.operator.operator import Operator
-from odl.space.pspace import ProductSpace
+from odl.operator import Operator
```

Comment:
This one was a misstake, i tried only changing them in other libraries where there is no such risk, as you mention.

Regarding "could not import diagnostics", yes that's it.